pkg/actuators/machine/validations.go

```diff
@@ -0,0 +1,169 @@
+	// UserDataSecret
+	if config.UserDataSecret == nil {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s UserDataSecret must be provided!", ErrorInvalidMachineObject))
+	} else if config.UserDataSecret.Name == "" {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s UserDataSecret *Name* must be provided!", ErrorInvalidMachineObject))
+	}
+
+	err := validateInstanceID(config)
+	if err != nil {
+		return err
+	}
+
+	// root disk of the node
+	if config.OSDisk == nil {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s OS Disk (os_disk) must be specified!", ErrorInvalidMachineObject))
+	}
+
+	err = validateVirtualMachineType(config.VMType)
+	if err != nil {
+		return err
+	}
+
+	if config.AutoPinningPolicy != "" {
+		err := autoPinningSupported(ovirtClient, config)
+		if err != nil {
+			return apierrors.InvalidMachineConfiguration(fmt.Sprintf("%s", err))
+		}
+	}
+	if err := validateHugepages(config.Hugepages); err != nil {
+		return apierrors.InvalidMachineConfiguration(fmt.Sprintf("%s", err))
+	}
+	return nil
+}
+
+// validateInstanceID execute validations regarding the InstanceID.
+// Returns: nil or InvalidMachineConfiguration
+func validateInstanceID(config *ovirtconfigv1.OvirtMachineProviderSpec) *apierrors.MachineError {
+	// Cannot set InstanceTypeID and at same time: MemoryMB OR CPU
+	if len(config.InstanceTypeId) != 0 {
+		if config.MemoryMB != 0 || config.CPU != nil {
+			return apierrors.InvalidMachineConfiguration(
+				fmt.Sprintf("%s InstanceTypeID and MemoryMB OR CPU cannot be set at the same time!", ErrorInvalidMachineObject))
+		}
+	} else {
+		if config.MemoryMB == 0 {
+			return apierrors.InvalidMachineConfiguration(
+				fmt.Sprintf("%s MemoryMB must be specified!", ErrorInvalidMachineObject))
+		}
+		if config.CPU == nil {
+			return apierrors.InvalidMachineConfiguration(
+				fmt.Sprintf("%s CPU must be specified!", ErrorInvalidMachineObject))
+		}
+	}
+	return nil
+}
+
+// validateVirtualMachineType execute validations regarding the
+// Virtual Machine type (desktop, server, high_performance).
+// Returns: nil or InvalidMachineConfiguration
+func validateVirtualMachineType(vmtype string) *apierrors.MachineError {
+	if len(vmtype) == 0 {
+		return apierrors.InvalidMachineConfiguration("VMType (keyword: type in YAML) must be specified")
+	}
+	switch vmtype {
+	case "server", "high_performance", "desktop":
+		return nil
+	default:
+		return apierrors.InvalidMachineConfiguration(
+			"error creating oVirt instance: The machine type must "+
+				"be one of the following options: "+
+				"server, high_performance or desktop. The value: %s is not valid", vmtype)
+	}
+}
+
+// validateHugepages execute validation regarding the Virtual Machine hugepages
+// custom property (2048, 1048576).
+// Returns: nil or error
+func validateHugepages(value int32) error {
+	switch value {
+	case 0, 2048, 1048576:
```

Comment:
These values should be moved to a const.

Reply:
Done